async/delay.ts

```diff
@@ -1,9 +1,13 @@
 // Copyright 2018-2021 the Deno authors. All rights reserved. MIT license.
 /* Resolves after the given number of milliseconds. */
-export function delay(ms: number): Promise<void> {
-  return new Promise((res): number =>
-    setTimeout((): void => {
+export function delay(ms: number, signal?: AbortSignal): Promise<void> {
+  return new Promise((res): void => {
+    const done = () => {
+      clearTimeout(i);
+      signal?.removeEventListener("abort", done);
       res();
-    }, ms)
-  );
+    };
+    const i = setTimeout(done, ms);
+    signal?.addEventListener("abort", done, { once: true });
```

Comment:
I think we new reject Promise when `abort` happening not call done

Reply:
Ok makes sense, I see `fetch` throws an `AbortError` so it does seem like it should be consistent with that, thanks.

Reply:
Glad to help

Reply:
Strangely it seems like AbortError isn't an actual error kind but I see this in deno itself:

https://github.com/denoland/deno/blob/main/extensions/fetch/26_fetch.js#L473

So I altered it to throw like that.